### PR TITLE
[Account Number] List Column ACH Transactions on Organization's Account Number page

### PR DESCRIPTION
**Pull request by rluodev**
We now list ACH transactions on each org's page (if they have an account number)

Screenshot 2024-06-03 at 7 31 40 PM

### CODE REVIEW

**Review by sampoder — COMMENTED** [review]
Some early feedback: I'd recommend paginating the transactions and it be nice if on the page there was a bit more context (like a heading or something)

File: app/views/events/account_number.html.erb [hunk 1]
```diff
@@ -69,6 +69,30 @@
...
+            </tr>
+          </thead>
+          <tbody data-behavior="transactions">
+            <%= render partial: "canonical_transactions/canonical_transaction", collection: @transactions, as: :ct, locals: { event: @event, show_amount: true, selectable: false } %>
+          </tbody>
+        </table>
+        <%= paginate @transactions %>
+        <% if @transactions.length == 0 %>
+          <p class="center">No Transactions Yet</p>
+        <% end %>
```
Comment (sampoder) [inline app/views/events/account_number.html.erb]: We should use the blankslate helper here

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -423,6 +423,10 @@ def sell_merch
   end
 
   def account_number
+    @transactions = CanonicalTransaction.where(transaction_source_type: "RawColumnTransaction", transaction_source_id: RawColumnTransaction.where("column_transaction->>'account_number_id' = '#{@event.column_account_number.column_id}'").pluck(:id))
+    page = (params[:page] || 1).to_i
+    @transactions = @transactions.page(page).per(25)
+    # TransactionGroupingEngine::Transaction::AssociationPreloader.new(transactions: @transactions, event: @event).run!
```
Comment (sampoder) [inline app/controllers/events_controller.rb]: What are we doing about this line?

Comment (rluodev) [reply app/controllers/events_controller.rb]: oh i'm just going to delete it lol

Comment (rluodev) [inline app/controllers/events_controller.rb]:
```suggestion
```

**Comment by rluodev** [comment]
all good for review!

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -423,6 +423,9 @@ def sell_merch
   end
 
   def account_number
+    @transactions = CanonicalTransaction.where(transaction_source_type: "RawColumnTransaction", transaction_source_id: RawColumnTransaction.where("column_transaction->>'account_number_id' = '#{@event.column_account_number.column_id}'").pluck(:id))
+    page = (params[:page] || 1).to_i
+    @transactions = @transactions.page(page).per(25)
```
Comment (sampoder) [inline app/controllers/events_controller.rb]:
```suggestion
    @transactions = @transactions.page(page).per(params[:per] || 25)
```

File: app/views/events/account_number.html.erb [hunk 4]
```diff
@@ -69,6 +69,30 @@
...
+        <table>
+          <thead>
+            <tr>
+              <th></th>
+              <th>Date</th>
+              <th>Description</th>
+              <th class="right-align">Amount</th>
+              <% if @show_running_balance %>
+                <th>Running Balance</th>
+              <% end %>
```
Comment (sampoder) [inline app/views/events/account_number.html.erb]: Let's not render the running balance

File: db/schema.rb [hunk 5]
```diff
@@ -52,9 +52,9 @@
     t.bigint "payment_recipient_id"
     t.string "recipient_email"
     t.boolean "send_email_notification", default: false
+    t.boolean "same_day", default: false, null: false
     t.string "company_name"
     t.string "company_entry_description"
-    t.boolean "same_day", default: false, null: false
     t.index ["column_id"], name: "index_ach_transfers_on_column_id", unique: true
```
Comment (sampoder) [inline db/schema.rb]: Let's revert this change

**Review by sampoder — DISMISSED** [review]
This looks good to me! Couple of things to clean up but then it should be good to merge

File: app/views/events/account_number.html.erb (hunk 4 above)
Comment (sampoder) [reply app/views/events/account_number.html.erb]:
```suggestion
```

**Comment by cjdenio** [comment]
Reverting because this breaks orgs that don't have an account number

**Comment by garyhtou** [comment]
@rluodev feel free to re-open this PR after handling the case where orgs don't have an account number generated yet!

**Comment by rluodev** [comment]
sounds good!
